caom2pipe/caom_composable.py

```diff
@@ -1109,8 +1108,8 @@ def update(self, observation, file_info):
         :param file_info: FileInfo instance
         :return:
         """
-        self._logger.debug(f'Begin update for {observation.observation_id}')
-        for plane in observation.planes.values():
+        self._logger.debug(f'Begin update for {self._observation.observation_id}')
```

Comment:
I couldn't see where this `update` function is used, but looking at the constructor suggests that the `self._observation` could be `None`.

Reply:
Here's where update is invoked:

- original call: https://github.com/opencadc/caom2pipe/blob/main/caom2pipe/caom_composable.py#L1207

- pull request call: https://github.com/SharonGoliath/caom2pipe/blob/CADC-12638/caom2pipe/caom_composable.py#L1207

The `self._observation` is assigned a value in either the constructor call at line 1165 of caom_composable.py, or after initialization, at line 1198 of caom_composable.py.

I don't think it can be None.